pkg/vault/test/authorizer/TimelockAuthorizerDelays.test.ts

```diff
@@ -0,0 +1,372 @@
+        expect(executableAt).to.equal((await currentTimestamp()).add(expectedExecutionDelay));
+      });
+
+      it('execution can be unprotected', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.false;
+      });
+
+      it('execution can be protected', async () => {
+        const executors = range(4).map(() => ethers.Wallet.createRandom().address);
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, executors, { from: root });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.true;
+        await Promise.all(
+          executors.map(async (executor) => expect(await authorizer.isExecutor(id, executor)).to.be.true)
+        );
+      });
+
+      it('root can cancel the execution', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+        expect(await authorizer.isCanceler(id, root)).to.be.true;
+
+        const receipt = await authorizer.cancel(id, { from: root });
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionCancelled', { scheduledExecutionId: id });
+      });
+
+      it('can be executed after the expected delay', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+
+        await advanceTime(expectedExecutionDelay);
+        const receipt = await authorizer.execute(id);
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', { scheduledExecutionId: id });
+      });
+
+      it('sets the new action delay when executed', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+
+        await advanceTime(expectedExecutionDelay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.delay(ACTION_1)).to.be.equal(ACTION_DELAY);
+      });
+
+      it('does not set any other action delay when executed', async () => {
+        const previousAction2Delay = await authorizer.delay(ACTION_2);
+
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+
+        await advanceTime(expectedExecutionDelay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.delay(ACTION_2)).to.be.equal(previousAction2Delay);
+      });
+
+      it('does not set any the grant action delay when executed', async () => {
+        const previousGrantDelay = await authorizer.getActionIdGrantDelay(ACTION_1);
+
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+
+        await advanceTime(expectedExecutionDelay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.delay(ACTION_2)).to.be.equal(previousGrantDelay);
```

Comment:
hmm shouldn't this be `ACTION_1`? I think I'm not getting why `delay(action2)` is compared with `grantDelay(action1)`.

Reply:
Looking at the test below, it should be the same action (1) and it should compare the current grant action delay with the old one.

Reply:
You are right!